cli/module_registry.go

```diff
@@ -209,28 +200,41 @@ func UploadModuleAction(c *cli.Context) error {
 				"if you want to upload a version without a meta.json, you must supply a module name and namespace (or module name and org-id)",
 			)
 		}
-		moduleID, err = updateManifestModuleIDWithArgs(c, client, nameArg, publicNamespaceArg, orgIDArg)
-		if err != nil {
-			return err
-		}
+
 	} else {
 		// if we can find a manifest, use that
+		var IdFromFile string
 		manifest, err := loadManifest(manifestPath)
 		if err != nil {
 			return err
 		}
 
-		moduleID, err = updateManifestModuleIDWithArgs(c, client, manifest.Name, publicNamespaceArg, orgIDArg)
-		if err != nil {
-			return err
+		if manifest.ModuleID != "" {
+			moduleID, err = parseModuleID(manifest.ModuleID)
+			if err != nil {
+				return err
+			}
+			IdFromFile = manifest.ModuleID
+		} else {
+			moduleID, err = parseModuleID(manifest.Name)
+			if err != nil {
+				return err
+			}
+			IdFromFile = manifest.Name
 		}
-		if nameArg != "" && nameArg != moduleID.name {
+
+		if nameArg != "" && nameArg != IdFromFile {
```

Comment:
```suggestion
		if nameArg != "" && nameArg != moduleID.Name {
```

This confusion is one of the reasons we want to do this PR `--name` should actually specify the module name (ex: `wifi-sensor`), not the module id ex: `fahmina:wifi-sensor` so this check should check against the `moduleName`, not the moduleID


I'm not sure you need the IdFromFile var

Reply:
isn't name the same as moduleID? like it would be fahmina:wifi-sensor on the name if the namespace is set?

Reply:
also name will be empty after we merge this change, so this may cause more bugs if we don't check which one is set?

Reply:
ah nvm i see what you mean! will fix